src/node/internal/streams_util.d.ts

```diff
@@ -1,3 +1,202 @@
+  resume?: Function;
+  write?: Function;
+  emit?: Function;
+  once?: Function;
+  removeListener?: Function;
+  destroy?: Function;
+  close?: Function;
+  abort?: Function;
+  listenerCount?: Function;
+  _construct?: Function;
+  _destroy?: Function;
+  setHeader?: Function;
+  socket?: any;
+  [kDestroyed]?: boolean;
+  [kIsErrored]?: boolean;
+  [kIsReadable]?: boolean;
+  [kIsDisturbed]?: boolean;
+}
+
+interface WebReadableStream {
+  pipeThrough: Function;
+  getReader: Function;
+  cancel: Function;
+}
+
+interface WebWritableStream {
+  getWriter: Function;
+  abort: Function;
+}
+
+// Stream detection functions
+export function isReadableNodeStream(obj: any, strict?: boolean): boolean;
+export function isWritableNodeStream(obj: any): boolean;
+export function isDuplexNodeStream(obj: any): boolean;
+export function isNodeStream(obj: any): boolean;
+export function isReadableStream(obj: any): boolean;
+export function isWritableStream(obj: any): boolean;
+export function isIterable(obj: any, isAsync?: boolean): boolean;
+
+// Stream state functions
+export function isDestroyed(stream: NodeStreamLike): boolean | null;
+export function isWritableEnded(stream: NodeStreamLike): boolean | null;
+export function isWritableFinished(
+  stream: NodeStreamLike,
+  strict?: boolean
+): boolean | null;
+export function isReadableEnded(stream: NodeStreamLike): boolean | null;
+export function isReadableFinished(
+  stream: NodeStreamLike,
+  strict?: boolean
+): boolean | null;
+export function isReadable(stream: NodeStreamLike): boolean | null;
+export function isWritable(stream: NodeStreamLike): boolean | null;
+
+interface IsFinishedOptions {
+  readable?: boolean;
+  writable?: boolean;
+}
+
+export function isFinished(
+  stream: NodeStreamLike,
+  opts?: IsFinishedOptions
+): boolean | null;
+export function isWritableErrored(stream: NodeStreamLike): Error | null;
+export function isReadableErrored(stream: NodeStreamLike): Error | null;
+export function isClosed(stream: NodeStreamLike): boolean | null;
+export function isOutgoingMessage(stream: any): boolean;
+export function isServerResponse(stream: any): boolean;
+export function isServerRequest(stream: any): boolean;
+export function willEmitClose(stream: NodeStreamLike): boolean | null;
+export function isDisturbed(stream: NodeStreamLike): boolean;
+export function isErrored(stream: NodeStreamLike): boolean;
+
+// Utility functions
+export const nop: () => void;
+export function once<T extends Function>(callback: T): T;
+
+// High water mark functions
+export function highWaterMarkFrom(
+  options: any,
+  isDuplex: boolean,
+  duplexKey: string
+): number | null;
+export function getDefaultHighWaterMark(objectMode?: boolean): number;
+export function setDefaultHighWaterMark(): never;
+export function getHighWaterMark(
+  state: { objectMode?: boolean },
+  options: any,
+  duplexKey: string,
+  isDuplex: boolean
+): number;
+
+// Abort signal function
+export function addAbortSignal(
```

Comment:
You should use as much as you can from types/node package. Declarations are available at https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/stream.d.ts

export { addAbortSignal } from 'node:stream'